src/external/wannier90.jl

```diff
@@ -0,0 +1,477 @@
+
+"""
+    Create a .win file for Wannier90, compatible with the system studied with DFTK.
+    Parameters to Wannier90 can be added as kwargs : e.g. num_iter = 500.
+"""
+function write_win(prefix::String, basis, ψ,
+                           num_wann::Integer;
+                           bands_plot=false,
+                           kwargs...
+                           )
+    open("$prefix.win", "w") do f
+
+        write(f, "! $prefix.win file generated by DFTK at $(now())\n\n")
+
+        # Required parameters
+        num_bands = size(ψ[1],2)
+        write(f,@sprintf("%-20s %s %5s \n","num_bands","=",num_bands))
+        write(f,@sprintf("%-20s %s %5s \n","num_wann","=",num_wann))
+        write(f,"\n")
+
+        # Optional parameters in kwargs
+        for (key,value) in kwargs
+            write(f,@sprintf("%-20s %s   %-30s\n",key,"=",value))
+        end
+        write(f,"\n\n")
+
+        # System
+        write(f,"!"^20*" System \n\n")
+
+        # Unit cell block
+        write(f,"begin unit_cell_cart\n"*"bohr\n")
+        unit_cell = transpose(basis.model.lattice) # cell vectors are in rows in Wannier90
+        for vec in eachrow(unit_cell)
+            write(f,@sprintf("%10.6f %10.6f %10.6f \n",vec[1],vec[2],vec[3]))
+        end
+        write(f,"end unit_cell_cart \n\n")
+
+        # Atoms block
+        atoms = basis.model.atoms
+        write(f,"begin atoms_frac\n")
+        for i in 1:length(atoms)                  # loop over different elements 
+            for coord in atoms[i][2]              # loop over atoms of the same element
+                write(f,@sprintf("%-2s %10.6f %10.6f %10.6f \n",
+                                 atoms[i][1].symbol,coord[1],coord[2],coord[3]) )
+            end
+        end
+        write(f,"end atoms_frac\n\n")
+
+        # k points
+        write(f,"!"^20*" k_points\n\n")
+
+        # Bands_plot block
+        if bands_plot
+            trash, labels, path = high_symmetry_kpath(basis.model)
+            write(f,"begin kpoint_path\n")
+            for i in 1:size(path[1],1)-1
+                # write path section A -> B
+                A,B = path[1][i],path[1][i+1]
+                (A == "\\Gamma") && (A = "Γ")
+                (B == "\\Gamma") && (B = "Γ")
+
+                # kpoints are writen in the file with normalized coordinates
+                k_A = round.(DFTK.normalize_kpoint_coordinate.(get!(labels,A,1)),
+                             digits = 5)
+                k_B = round.(DFTK.normalize_kpoint_coordinate.(get!(labels,B,1)),
+                             digits = 5)
+                write(f,@sprintf("%s %10.6f %10.6f %10.6f  %s %10.6f %10.6f %10.6f \n",
+                                 A,k_A[1],k_A[2],k_A[3],B,k_B[1],k_B[2],k_B[3]) )
+            end
+            write(f,"end kpoint_path\n")
+            write(f,"bands_plot = T\n\n")
+        end
+
+        # Mp grid
+        write(f,"mp_grid : $(basis.kgrid[1]) $(basis.kgrid[2]) $(basis.kgrid[3])\n\n")
```

Comment:
kshift is not used anywhere? Does this assume that kshift == [0,0,0]? If yes, do it explicitly

Reply:
Indeed I forgot to write the shift in the input file... Thanks ! 
I just have to add it to each kpoint line. I tested it, it doesn't change anything in the case of Silicon so that's why I didn't see it... If "kshift = [y1 y2 y3]" I just have to replace each kcoords line "x1 x2 x3" by:
"x1 x2 x3 y1 y2 y3"

Reply:
Nevermind when I run through all `kpoints.coordinate` to write in the win file, it seems that the given coordinates already take shift into account. So no need to shift right ?

Reply:
Ah right, since the list is explicit probably you don't need to do anything, OK!